### ISSUE TITLE
Reps Analytics not complete

### ISSUE THREAD

**Issue by MichaelKohler** [mozilla/participation-metrics-org#181]
I've noticed that the Reps Analytics (Events and Activities) are not complete for the past few months. This might be due to a recent Portal deployment, happy to help out if so.

There seem to be spikes where it should be mostly continous (with some fluctuation of course):

<img width="568" alt="screen shot 2018-07-21 at 13 34 12" src="https://user-images.githubusercontent.com/330324/43035156-cfd4b288-8cea-11e8-86cd-2485bde176ba.png">
<img width="571" alt="screen shot 2018-07-21 at 13 34 26" src="https://user-images.githubusercontent.com/330324/43035157-cfef0264-8cea-11e8-920a-e5e025c93110.png">

# Activities

There don't seem to be any reports between July 20th and July 13th:

<img width="1119" alt="screen shot 2018-07-21 at 13 25 07" src="https://user-images.githubusercontent.com/330324/43035124-4315b50e-8cea-11e8-83cd-bd8958408954.png">

However there are activities in the data source at https://reps.mozilla.org/reports/:

<img width="1011" alt="screen shot 2018-07-21 at 13 28 23" src="https://user-images.githubusercontent.com/330324/43035129-5bdf511c-8cea-11e8-8174-fdd1b87d7788.png">

# Events

The same goes for events:

<img width="555" alt="screen shot 2018-07-21 at 13 32 36" src="https://user-images.githubusercontent.com/330324/43035138-92e385ac-8cea-11e8-85de-8ea771615f5c.png">

However there are activities in the data source at https://reps.mozilla.org/events/#/period/past/:

<img width="1010" alt="screen shot 2018-07-21 at 13 33 29" src="https://user-images.githubusercontent.com/330324/43035149-ad42cc3c-8cea-11e8-8ba2-1170a9485a79.png">

As seen in the initial screenshots, these timeframes are not the only one affected, I've just analyzed this one.

Would be great if this could be fixed. As said, I'm happy to help debug this if needed. Kudos to @freaktechnik for making me aware of this.

**Comment by Mte90** [comment]
Very strange, we didn't changed the rest api since a while so maybe it is a timeout?
We need an error log from bitergia to see the problem

**Comment by mafesan** [comment]
Hi @MichaelKohler and @Mte90,

I'm Miguel-Angel, from Bitergia support team. Thank you for reporting this issue and sorry about the delay with it. Our Ops team has re-generated the database indexes and the data looks good now. However, our team is still investigating what caused this error. We'll keep you updated as soon as we have more information about it.

Best, Miguel-Angel

**Comment by MichaelKohler** [comment]
Definitely looks better now! The bar charts seem ok (didn't do a full check). However, the "Last events" widget still is missing some data - which is present in the bar chart.

<img width="634" alt="screen shot 2018-07-24 at 12 04 43" src="https://user-images.githubusercontent.com/330324/43131774-22072cce-8f3a-11e8-976c-85e5f87098d0.png">

**Comment by mafesan** [comment]
@MichaelKohler Our team has reviewed this again, we think there is a difference in the data because each of these events has a starting and an ending date: in the dashboard appears the starting date of each event (the list in your screen-shot is sorted by this field), while in https://reps.mozilla.org/events/#/period/past/ the list is sorted using the ending date. 

For instance, if you look at the `Moz Stall @ Angel Hack` event in your website, 

![reps-last-events](https://user-images.githubusercontent.com/9032790/43211017-ded38108-9030-11e8-8e2b-e87c10e7b7f9.png)

You will see that event both in the bar-chart and the `Last Events` widget marked with the starting date (Jun 30th):

![reps-event-angel](https://user-images.githubusercontent.com/9032790/43211210-4dfdc070-9031-11e8-9d6c-856210bd525d.png)

(You can check this event in https://analytics.mozilla.community/goto/68136e73c00b0a1e870ec97a692fad1a, which appears on the bar-chart too).

What do you think about this? Thank you!

**Comment by MichaelKohler** [comment]
OK, so far so good. I still see for example this gap though (and I've sorted by start date):

<img width="640" alt="screen shot 2018-07-25 at 17 50 39" src="https://user-images.githubusercontent.com/330324/43212053-537e0d5a-9033-11e8-9f96-75edc7b5b99e.png">

There is no event listed between June 22nd and June 29th.

For example:

<img width="1014" alt="screen shot 2018-07-25 at 17 53 00" src="https://user-images.githubusercontent.com/330324/43212233-ba7725dc-9033-11e8-8879-61a12079b4c6.png">

Let's take the "Security Analytics Workshop" entry. This has a starting and end date of June 23rd. I can also see it when filtering in Kibana: https://analytics.mozilla.community/goto/5466e58dc75c30e213f17f72414fd86e

However, without any filter, I don't see it showing up in "Last events" (see first screenshot). There is something wrong, I can't pinpoint it though.

**Comment by mafesan** [comment]
@MichaelKohler You are absolutely right, thank you for providing us that example. Due to how the `Last Events` widget is built, we set a limit to display only 30 results in it, in order to increase the performance of the dashboard. Instead of displaying the last 30 events, it seems Kibana has skipped some of the results randomly (only in June there were 89 events), so we have opted for increasing this limit to 100 and now you should see this data in the widget.

Please, can you check it again? Thank you and sorry for the inconveniences.

(See https://analytics.mozilla.community/edit/goto/10fb8ba0c0b3d3005b35ec259fdb0fae)
![reps-limit-change](https://user-images.githubusercontent.com/9032790/43213724-9f4a5abe-9037-11e8-9af6-6aef551676b3.png)

**Comment by MichaelKohler** [comment]
Ah, got it. This looks way better now. However, I think the Order does not work yet. I have to manually sort by start date to get them in descending order:

<img width="642" alt="screen shot 2018-07-25 at 18 29 40" src="https://user-images.githubusercontent.com/330324/43214195-dd5f4b92-9038-11e8-94a0-8aa78f27e9cd.png">
<img width="639" alt="screen shot 2018-07-25 at 18 30 27" src="https://user-images.githubusercontent.com/330324/43214196-dd774ab2-9038-11e8-8c75-2f6d571525f0.png">

**Comment by sanacl** [comment]
Having a look at this table and the metrics it has, the displayed date is the max `start_date` while the sorting is using `start_date`. We are working on it. Thanks for the feedback @MichaelKohler

**Comment by sanacl** [comment]
The issue we have with the table is related to the way Kibana display the information. The visualization by default groups by 'event title' and sorts by 'start date' instead of using only start date to sort all the events. This is why we get this strange behavior by default (until you click on the start date). The most simple way of changing this behavior is to change the order of the columns so the first one is unique ('event title' is not unique). 

Due to this widget is used by you, we would love to get your feedback about the proposed changes before applying them to the standard panel. Are u ok with the changes?

Old one: https://analytics.mozilla.community/edit/goto/7b3381ab5f5503c8d35abbd623fe69a1
Proposed one: https://analytics.mozilla.community/edit/goto/ad6ed370089e6e2e9ca0d00cf9195783

Thank u :)

**Comment by MichaelKohler** [comment]
I'd be fine with this.

**Comment by hmitsch** [comment]
@Mte90 are you also okay?
Once we have @MichaelKohler and you agree to this, I'd request @sanacl to move ahead with the proposed change and apply them to the standard panel.

-Henrik

**Comment by Mte90** [comment]
I am fine too!

**Comment by sanacl** [comment]
This is done. I've also fixed the link on the right table, which was not properly displayed.

![canvas](https://user-images.githubusercontent.com/326719/43555693-1c4d9714-95fc-11e8-9b68-4c3a0f137b7d.png)
